examples/server/utils.hpp

```diff
@@ -133,26 +134,50 @@ struct completion_token_output
     std::string text_to_send;
 };
 
-static inline void server_log(const char *level, const char *function, int line,
-                       const char *message, const nlohmann::ordered_json &extra)
+static void server_log(const char *level, const char *function, int line, const char *message, const nlohmann::ordered_json &extra)
 {
-    nlohmann::ordered_json log
-    {
+    std::stringstream ss_tid;
+    ss_tid << std::this_thread::get_id();
+    json log = nlohmann::ordered_json{
+        {"tid", ss_tid.str()},
         {"timestamp", time(nullptr)},
-        {"level",     level},
-        {"function",  function},
-        {"line",      line},
-        {"message",   message},
     };
 
-    if (!extra.empty())
-    {
-        log.merge_patch(extra);
-    }
+    if (server_log_json) {
+        log.merge_patch(
+                {
+                        {"level",     level},
+                        {"function",  function},
+                        {"line",      line},
+                        {"msg",       message},
+                });
+        if (!extra.empty()) {
+            log.merge_patch(extra);
+        }
+
+        const std::string str = log.dump(-1, ' ', false, json::error_handler_t::replace);
+        printf("%.*s\n", (int)str.size(), str.data());
+        fflush(stdout);
+    } else {
+        char buf[1024];
+        snprintf(buf, 1024, "%24s %4s: %-80s", function, level, message);
```

Comment:
Personally, I find that `%-80` is quite long, it makes the log quite hard to read on small screen:

```
            update_slots VERBOSE: Slots updated                                                                    tid="140566093502336" timestamp=1708812445
              start_loop VERBOSE: wait for new task                                                                tid="140566093502336" timestamp=1708812445
```

Also, log level should show first I think. A format like `LEVEL [function_name] message | additional=data abc=def` would be preferable.

Another thing to notice is that the log line is now limited to 1024 characters, personally for verbose log I prefer to log everything even if it's long.

I understand that the `snprintf` is used here for padding in table-like, but I'd prefer either using `std::stringstream` (no padding), either `snprintf`, but not both.

Reply:
Agreed with the expected format you proposed.
@ggerganov could we go with: `LEVEL [function_name] message | additional=data `

Reply:
Yes